suite/tests/client-interface/mangle_suspend.c

```diff
@@ -60,22 +63,28 @@ suspend_thread_routine(void *arg)
     /* This thread is executing labels for the client to insert a clean call that
      * does the suspend and subsequent check for correctness.
      */
+    int suspend_val = *(int *)arg;
     while (!test_ready) {
         /* Empty. */
     }
-    struct timespec sleeptime;
-    sleeptime.tv_sec = 0;
-    /* Pick an uneven number to maximize the number of suspend calls w/o
-     * running into limitations in DR.
-     */
-    sleeptime.tv_nsec = 2000 * 1111;
     while (!test_done) {
-        asm volatile("mov %0, %%rdx\n\t"
-                     "mov %0, %%rdx\n"
-                     :
-                     : "i"(SUSPEND_VAL_C)
-                     : "rdx");
-        nanosleep(&sleeptime, NULL);
+        if (suspend_val == SUSPEND_VAL_TEST_1_C) {
+            asm volatile("mov %0, %%rdx\n\t"
+                         "mov %0, %%rdx\n"
+                         :
+                         : "i"(SUSPEND_VAL_TEST_1_C)
+                         : "rdx");
+        } else {
```

Comment:
I think it would be cleaner to have two separate functions rather than one with split behavior.

Reply:
Done.